docs/guides/deployment/deployment.md

```diff
@@ -0,0 +1,106 @@
+# Deployment
+
+Applications built with Platformatic DB can be deployed to a hosting service
+in the same way as any other Node.js application. This guide covers a few
+things that will help smooth the path from development to production.
+
+## Server configuration
+
+:::info
+
+See the [Configuration](/reference/configuration.md) reference for all
+configuration settings.
+
+:::
+
+### Configuration with environment variables
+
+We recommend that you use [environment variable placeholders](/reference/configuration.md#environment-variable-placeholders)
+in your Platformatic DB configuration. This will allow you to configure
+different settings in your development and production environments.
+
+In development you can set the  environment variables via a **`.env`** file
+that will be automatically loaded by Platformatic DB. For example:
+
+```
+PORT=3042
+PLT_SERVER_HOSTNAME=127.0.0.1
+```
+
+In production your hosting provider will typically provide their own mechanism
+for setting environment variables.
+
+### Configure the server port
+
+Configure the port that the server will listen on by setting an environment
+variable placeholder in your Platformatic DB configuration file:
+
+```json title="platformatic.db.json"
+{
+  "server": {
+    ...
+    "port": "{PORT}"
+  },
+  ...
+}
+```
+
+### Listen on all network interfaces
+
+Most hosting providers require that you configure your server to bind to all
+available network interfaces. To do this you must set the server hostname to
+`0.0.0.0`.
+
+This can be handled with an environment variable placeholder in your Platformatic
+DB configuration file:
+
+```json title="platformatic.db.json"
+{
+  "server": {
+    ...
+    "hostname": "{PLT_SERVER_HOSTNAME}",
+  },
+  ...
+}
+```
+
+The environment variable `PLT_SERVER_HOSTNAME` should then be set to `0.0.0.0`
+in your hosting environment.
+
+### Security considerations
+
+We recommend disabling the GraphiQL web UI in production. It can be disabled
+with the following configuration:
+
+```json title="platformatic.db.json"
+{
+  "core": {
+    ...
+    "graphql": {
+      "graphiql": false
+    }
+  },
+  ...
+}
+```
+
+If you want to use this feature in development, replace the configuration
+values with [environment variable placeholders](/reference/configuration.md#environment-variable-placeholders)
+so you can set it to `true` in development and `false` in production.
+
+## Databases
+
+### Applying migrations
+
+TODO: When should migrations be run in production? During deployment?
```

Comment:
When should migrations be run in production? During deployment? Probably depends on the DBMS, as well as how many application instances are running.